### PR TITLE
Use compile time optimization selection for multi-arch

### PR DESCRIPTION
**Pull request by khourig**
For multi-arch, CMAKE_SYSTEM_PROCESSOR is empty. At compile time,
the processor is detected and the optimized functions are called.
The choice is either remove all optimization or include all optimized
source code with irrelavant sections removed by compile time
architecture detection. This change uses the latter option.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -82,27 +101,28 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "^arm" OR
     message(FATAL_ERROR
             "PNG_ARM_NEON must be one of [${PNG_ARM_NEON_POSSIBLE_VALUES}]")
   elseif(NOT ${PNG_ARM_NEON} STREQUAL "off")
-    set(libpng_arm_sources
```
Comment (ruslo) [inline CMakeLists.txt:85]: Why has it to be moved? For minimal patch purposes, it should be kept here.

**Review by ruslo — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (khourig) [reply CMakeLists.txt:85]: The hardware specific code should always be included, even if it is not for the target hardware. The code will preprocess itself away if it does not match the current target hardware. For iOS, multiple hardware is targeted so `CMAKE_SYSTEM_PROCESSOR` is not set and **no** optimized source was included.

Comment (ruslo) [reply CMakeLists.txt:85]: > For iOS, multiple hardware is targeted so CMAKE_SYSTEM_PROCESSOR is not set and no optimized source was included

For combined iOS build we will enter this `if` too, see `OR CMAKE_SYSTEM_PROCESSOR STREQUAL ""` condition.

**Review by ruslo — CHANGES_REQUESTED** [review]

**Comment by ruslo** [comment]
https://github.com/ruslo/hunter/blob/v0.23.214/README.rst

**Pull request closed without merging**